test/test_productbasis.jl

```diff
@@ -0,0 +1,95 @@
+using Test
+using Polynomials4ML.Testing: println_slim, print_tf
+using Polynomials4ML: ProductBasis, evaluate, test_evaluate
+using LinearAlgebra: norm
+using BenchmarkTools
+using Polynomials4ML
+using ACEbase.Testing: fdtest
+using Printf
+
+
+N1 = 10
+N2 = 20
+N3 = 30
+
+B1 = randn(N1)
+B2 = randn(N2)
+B3 = randn(N3)
+
+spec = sort([ (rand(1:N1), rand(1:N2), rand(1:N3)) for i = 1:100 ])
+
+basis = ProductBasis(spec)
+
+
+## 
+
+@info("Test serial evaluation")
+
+BB = (B1, B2, B3)
+
+A1 = test_evaluate(basis, BB)
+A2 = evaluate(basis, BB)
+
+println_slim(@test A1 ≈ A2 )
+
+##
+
+@info("Test batch evaluation")
+
+nX = 64 
+bBB = ( randn(nX, N1), randn(nX, N2), randn(nX, N3) )
+bA1 = zeros(ComplexF64, nX, length(basis))
+
+for j = 1:nX
+    bA1[j, :] = evaluate(basis, (bBB[1][j, :], bBB[2][j, :], bBB[3][j, :]))
+end
+
+bA2 = evaluate(basis, bBB)
+
+println_slim(@test bA1 ≈ bA2)
+
+
+## 
+
+@info("Testing _prod_grad")
+
+using StaticArrays, ForwardDiff
+
+prodgrad = Polynomials4ML._prod_grad
+
+for N = 1:5 
+   for ntest = 1:10
+      local v1, g 
+      b = rand(SVector{3, Float64})
+      g = prodgrad(b.data, Val(3))
```

Comment:
should this have a Val(N) ?

Reply:
Yes I think so. I didn't aware that when I was copying it from https://github.com/ACEsuit/ACEcore.jl/blob/main/test/test_1pbasis.jl.

Thank you for reviewing.